### DISCUSSION TITLE
need a specific body

### DISCUSSION THREAD

**Discussion by MherMg** [relayrides/pushy#993] [Q&A]
we need to have such a body, but there is no possibility, what do you suggest

```json
{
    "aps": {
        "sound": "",
        "loc-key": "<alert-msg-id>",
        "loc-args": ["<from>"],
        "call-id": "<call-id>",
        "uuid": "<uid>",
        "send-time": "<date>"
    },
    "from-uri": "<from-uri>",
    "display-name": "<from-name>",
    "pn_ttl": <ttl>,
    "customPayload": {}
}
```

**Comment by jchambers** [comment]
I think [`ApnsPayloadBuilder#addCustomProperty(String, Object)`](https://pushy-apns.org/apidocs/latest/com/eatthepath/pushy/apns/util/ApnsPayloadBuilder.html#addCustomProperty(java.lang.String,java.lang.Object)) will do what you need. If not, remember that a payload is just a JSON string, and you can construct it however you want; you're not required to use the payload builder included with Pushy.

**Reply by MherMg** [reply]
when I add custom property structure JSON looks like this

```json
{
    "aps": {
    },
    "custom1": "string",
    "custom2": "string"
}
```

but I need like this

```json
{
    "aps": {
     "custom1": "string",
     "custom2": "string"
    }
}
```

**Reply by jchambers** [reply]
Generally speaking, custom properties are not allowed inside the `aps` object, and consequently Pushy doesn't provide tools to add custom properties to the `aps` object. If this what you want to do, I'm afraid you'll need to build payloads on your own.
